### PR TITLE
core: allow bid adapters to return null fledgeAuctionConfigs

### PR DESCRIPTION
**Pull request by dgirardi**
## Type of change

- [x] Bugfix

## Description of change

Allow adapters to return an object with null/undefined paapi configs (e.g. `{bids: [...]}`), to avoid boilerplate that unwraps return values from ortbConverter.

### CODE REVIEW

**Review by robertrmartinez — APPROVED** [review]
Is there a plan to migrate to something different than `fledgeAuctionConfigs` at all?

Not involved in this project much but I know the name `fledge` has gone away.

Should we add support for a key `paapiConfigs`?

File: src/adapters/bidderFactory.js [hunk 1]
```diff
@@ -361,6 +361,18 @@ export function newBidder(spec) {
...
+// TODO: remove this in prebid 9
+const PAAPI_RESPONSE_PROPS = ['paapiAuctionConfigs', 'fledgeAuctionConfigs'];
+const RESPONSE_PROPS = ['bids'].concat(PAAPI_RESPONSE_PROPS);
+function getPaapiConfigs(adapterResponse) {
+  const [paapi, fledge] = PAAPI_RESPONSE_PROPS.map(prop => adapterResponse[prop]);
+  if (paapi != null && fledge != null) {
+    throw new Error(`Adapter response should use ${PAAPI_RESPONSE_PROPS[0]} over ${PAAPI_RESPONSE_PROPS[1]}, not both`);
+  }
+  return paapi ?? fledge;
+}
```
Comment (robertrmartinez) [inline src/adapters/bidderFactory.js:374]: niiiiice

File: test/spec/unit/core/bidderFactory_spec.js [hunk 2]
```diff
@@ -1456,67 +1456,85 @@ describe('bidderFactory', () => {
...
-          bidder.callBids(bidRequest, addBidResponseStub, doneStub, ajaxStub, onTimelyResponseStub, wrappedCallback);
+        ['fledgeAuctionConfigs', 'paapiAuctionConfigs'].forEach(paapiProp => {
+          describe(`using ${paapiProp}`, () => {
+            it('should call paapi hook with PAAPI configs', function() {
+              const bidder = newBidder(spec);
+              spec.interpretResponse.returns({
+                bids: bids,
+                [paapiProp]: [paapiConfig]
+              });
+              bidder.callBids(bidRequest, addBidResponseStub, doneStub, ajaxStub, onTimelyResponseStub, wrappedCallback);
```
Comment (robertrmartinez) [inline test/spec/unit/core/bidderFactory_spec.js:1512]: maybe a test for `prefers 'paapiAuctionConfigs' over 'fledgeAuctionConfigs' if both defined`?

Prob not a real life scenario though so maybe not!

**Review by robertrmartinez — APPROVED** [review]
LGTM

**Review by Rothalack — APPROVED** [review]
Looks good!

**Pull request merged**